### ISSUE TITLE
Sedml figures are not correctly generated

### ISSUE THREAD

**Issue by bilalshaikh42** [reproducible-biomedical-modeling/CRBM-Viz#2655]
The frontend is not able to display the 2D plots defined in the Sedml. This is due to a mismatch between the ids for the outputs of the simulation run as defined in the HDF file and that returned by the combine service. [Here](https://run.biosimulations.dev/simulations/60d9deb771d8706e02beace7#tab=design-viz) is the example run that the following references

The root of the problem is here: 

 The combine service returns the following information about  one of  the curves present in a sedml file: 
```
id: "curve_Figure_3c_Swe1M"
name: "Modified Swe1"
xData: "simulation_1.sedml/Figure_3c/data_gen_time"
yData: "simulation_1.sedml/Figure_3c/data_gen_Swe1M"
```

The results service returns the following truncated map of the ids and data for the run:
```
.....
data_set_PClb: {id: "data_set_PClb", location: "simulation_1.sedml", reportId: "report", label: "Phosphorylated Clb2", values: 
__data_set__Figure_3c_data_gen_PSwe1M: {id: "__data_set__Figure_3c_data_gen_PSwe1M", location: "simulation_1.sedml", reportId: "Figure_3c", label: "data_gen_PSwe1M", values: Array(201)}
__data_set__Figure_3c_data_gen_Swe1: {id: "__data_set__Figure_3c_data_gen_Swe1", location: "simulation_1.sedml", reportId: "Figure_3c", label: "data_gen_Swe1", values: Array(201)}
__data_set__Figure_3c_data_gen_Swe1M: {id: "__data_set__Figure_3c_data_gen_Swe1M", location: "simulation_1.sedml", reportId: "Figure_3c", label: "data_gen_Swe1M", values: Array(201)}
__data_set__Figure_3c_data_gen_Swe1T: {id: "__data_set__Figure_3c_data_gen_Swe1T", location: "simulation_1.sedml", reportId: "Figure_3c", label: "data_gen_Swe1T", values: Array(201)}
__data_set__Figure_3c_data_gen_time: {id: "__data_set__Figure_3c_data_gen_time", location: "simulation_1.sedml", reportId: "Figure_3c", label: "data_gen_time", values: Array(201)}
__data_set__Figure_3d_data_gen_time: {id: "__data_set__Figure_3d_data_gen_time", location: "simulation_1.sedml", reportId: "Figure_3d", label: "data_gen_time", values: Array(201)}

......
```
The mismatch is happening with the ids of the reports. Currently, the frontend is checking for ```simulation_1.sedml/Figure_3c/data_gen_time``` to be present in the list, but is unable to find it, since data from the figures is prepended with the ```__data_set__Figure_3c_``` string. The data is retrieved from the HDF file. I confirmed that the file has the same ids set as the attributes, so this might be an issue with the simulators.  The [conventions](https://biosimulators.org/conventions/simulation-reports) say the following about the id labels: 

> Ids of rows (SED data sets or data generators): For reports, the ids of the data sets should be encoded into the key sedmlDataSetIds. The value of this key should be an array of the ids of the data sets, in the order in which the data sets were defined in their parent SED document. For plots, the ids of the data generators should be encoded into the key sedmlDataSetIds. The value of this key should be an array of the ids of the data generators, in the order in which the data generators were defined in their parent SED document.

From my reading, the datasets should have the id "data_gen_time".  I assume the prefix has been added to distinguish the same generator used in multiple plots. However, this should not be neccesarry as any refrences to a particular generator by id will need to have a specific output id before it. eg ```  simulation_1.sedml/Figure_3c/data_gen_time``` vs ```simulation_1.sedml/Figure_3d/data_gen_time```  If we want, we can actually have each of the data generators be its own dataset, and then use the HDF5 links to have multiple figures point to the same dataset (derived from the datagenerator). I think this would better preserve the structure of the sedml, but am not sure if it gives much benefits beyond that. 

@jonrkarr Is this a change that needs to be made to the simulators?

**Comment by bilalshaikh42** [comment]
@jonrkarr  I am a bit confused about the outputs of the simulators here, as they seem to be inconsistent.

**Comment by jonrkarr** [comment]
The dispatch app had been using labels rather than ids because only the labels were encoded into CSV files. Now that the HDF5 files have both labels (which aren't required to be unique) and ids (which must be unique), we can switch to using ids to link SED-ML descriptions of plots and their data.

**Comment by jonrkarr** [comment]
The code for mapping results to SED-ML plots had a couple of todos associated with dealing with ids vs labels. I took care of that. The plots seem to be working now, except for the simulations with tellurium (see below), VCell (you're working on this), GillesPy2 (see #2657), and #2652.

The issue above with extra prefix is specifically an issue with tellurium. tellurium is implemented a little differently so it can do some of its own SED-ML processing. I corrected this to work the same as the other images. This seems to be working now. Example: https://run.biosimulations.dev/simulations/60dbfe52e041030305f7d3cb.

I also updated the Vega examples to work with the new HDF5 data service.
